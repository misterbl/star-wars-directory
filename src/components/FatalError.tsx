import React from "react";
import ROUTES from "../const/routes";
import BackButton from "./BackButton";

interface IFatalError {
  push: (path: string, state?: any) => void;
  resetFatalError: () => void;
}

const FatalError = ({ push, resetFatalError }: IFatalError) => (
  <div className="fatal-error">
    <BackButton
      text="Back home"
      route={ROUTES.INDEX}
      push={push}
      onClick={resetFatalError}
    />
  </div>
);

export default FatalError;
