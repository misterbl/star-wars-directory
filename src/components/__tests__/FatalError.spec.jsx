/* eslint-disable no-undef */
import * as React from "react";
import { shallow } from "enzyme";
import FatalError from "../FatalError";

describe("FatalError", () => {
  const props = {
    push: jest.fn(),
    resetFatalError: jest.fn()
  };
  const wrapper = shallow(<FatalError {...props} />);
  it("matches the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("calls resetFatalError() onClick ", () => {
    const backButton = wrapper.find("BackButton").at(0);
    backButton.simulate("click");
    expect(props.resetFatalError).toHaveBeenCalled();
  });
});
