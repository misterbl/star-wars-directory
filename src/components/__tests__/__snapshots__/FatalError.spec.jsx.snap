// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FatalError matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FatalError
    push={[MockFunction]}
    resetFatalError={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <BackButton
        onClick={[MockFunction]}
        push={[MockFunction]}
        route="/"
        text="Back home"
      />,
      "className": "fatal-error",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "onClick": [MockFunction],
        "push": [MockFunction],
        "route": "/",
        "text": "Back home",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <BackButton
          onClick={[MockFunction]}
          push={[MockFunction]}
          route="/"
          text="Back home"
        />,
        "className": "fatal-error",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onClick": [MockFunction],
          "push": [MockFunction],
          "route": "/",
          "text": "Back home",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
